### PR TITLE
feature #573: created endpoint for general events for day period

### LINKED ISSUE #573: Endpoint for general events for day period (tg)
For Telegram bot, you need to create an endpoint for returning events. You need to create an endpoint that returns all general group events for a specific day or the current day.

I remind you that general events are events that are available to unregistered users, that is, they are ordinary pairs.

### PR DESCRIPTION
**Pull request by SharpDevOps10**
feature #573: created endpoint for general events for day period

- added method for getting special day
- added method for getting general events by day
- created endpoint

closes #573

### CODE REVIEW

**Review by Hahhhayam — CHANGES_REQUESTED** [review]
- no validation in day parameter
image

- by some reason, your method works with day > 6
image

- method should works withot week parameter (if it undefiend, set current)
image

File: src/v2/api/services/ScheduleService.ts [hunk 1]
```diff
@@ -81,13 +105,35 @@ export class ScheduleService {
...
+    const filteredEvents = events.filter((event) => {
+      const indexOfLesson = this.getIndexOfLesson(event.period, event.startTime, endOfWeek);
+      return indexOfLesson !== null;
+    });
+
+    return await Promise.all(
+      filteredEvents.map(async (event) => {
+        const { startWeek, endWeek } = await this.setWeekTime(event, week);
+        const discipline = await this.getEventDiscipline(event.id);
+        return { event, discipline, startWeek, endWeek };
```
Comment (f-ogoog) [inline src/v2/api/services/ScheduleService.ts:117]: it should return only events and week, not discipline and dates of week

File: src/v2/api/services/ScheduleService.ts [hunk 2]
```diff
@@ -49,17 +50,40 @@ export class ScheduleService {
...
+  }
+
+  async getGeneralGroupEvents (id: string, week: number) {
+    const currentSemester = await this.dateService.getCurrentSemester();
+    const weekToCheck = week || (!currentSemester.isFinished ? await this.dateService.getCurrentWeek() : null);
+    if (weekToCheck === null || !this.isWeekValid(weekToCheck, currentSemester)) {
+      throw new InvalidWeekException();
+    }
     const { startOfWeek, endOfWeek } = week ? await this.dateService.getDatesOfWeek(week) : this.dateService.getDatesOfCurrentWeek();
+
```
Comment (f-ogoog) [inline src/v2/api/services/ScheduleService.ts:86]: please, add in findMany field period, and that field can not be NO_PERIOD

Comment (kropli) [reply src/v2/api/services/ScheduleService.ts:86]: resolved in https://github.com/fictadvisor/fictadvisor-api/pull/564

**Review by kropli — APPROVED** [review]

**Pull request merged**